### PR TITLE
C37 event log on actions

### PR DESCRIPTION
**Pull request by frinzekt**
## Change Summary
- added `QueueEventLog` model
- added functionality to populate `QueueEventLog` on the existing API to add/update Queue entry

### Change Form
**Fill this up (NA if not available). If a certain criteria is not met, can you please give a reason.**

- [x] The pull request title has an issue number
- [x] The code works by "Smoke testing" (in other words, you have tested it your own and it works)
- [ ] The code has documentation (either comments or in main documentation platform)

## Other Information
**[Is there anything in particular in the review that I should be aware of?]**

### CODE REVIEW

File: app/app/queue.py [hunk 1]
```diff
@@ -11,19 +11,24 @@
...
+        db.session.commit()
+
+        # new_queue_entry only has the id after db commit
+        new_queue_event_log = QueueEventLog(new_queue_entry)
+        db.session.add(new_queue_event_log)
         db.session.commit()
-        print(new)  # This print is important (do not remove)
-        return new.to_dict(), 201
+
+        print(new_queue_entry)  # This print is important (do not remove) - Unknown reason (further investigation needed)
```
Comment (JordanHartley) [inline app/app/queue.py:30]: Also very curious why these prints are needed

Comment (frinzekt) [reply app/app/queue.py:30]: This prints are necessary because if it doesn't exist, the endpoint doesn't return.

Why? Flask bug maybe?

Comment (frinzekt) [reply app/app/queue.py:30]: thats why it needs more investigation

**Review by JordanHartley — APPROVED** [review]
All looks good to me

**Comment by frinzekt** [comment]
Closed as not needed anymore

**Pull request closed without merging**